src/coreclr/inc/sstring.h

```diff
@@ -125,8 +125,7 @@ class EMPTY_BASES_DECL SString : private SBuffer
     static const SString &Empty();
 
     SString();
-
-    explicit SString(const SString &s);
```

Comment:
What are all existing places that use this constructor? Can they be broken by changing what this constructor does?

Are there places that will unexpectedly start using this constructor with removal of `explicit` (and may get broken)?

I am wondering whether it would be better to make this explicit so one can tell by looking at the callsites what is exactly going on.

Reply:
> Can they be broken by changing what this constructor does?

Yes they can. It is a huge mess to detangle this in general. That is why this PR is draft. I wanted to see across systems quickly how this behaves. I am still looking into approaches that will make this work in an intuitive way.

Reply:
I've convert to move ctor/assignment.